rasterio/_err.pyx

```diff
@@ -216,6 +216,8 @@ cdef void *exc_wrap_pointer(void *ptr) except NULL:
         if exc:
             raise exc
     CPLErrorReset()
+    if ptr == NULL:
+        raise SystemExit("Fatal Error: NULL pointer returned without error.")
```

Comment:
I'm not sure about this, it would be hard to recover from?

Reply:
This prevents crashing as the `except NULL` expects an exception to be raised when NULL is returned.

Reply:
I don't believe this is necessary for this PR. It was something I added when I ran into this error while fixing things. I removed it and created an issue to discuss #2492.